lib/sync/endpoint.php

```diff
@@ -0,0 +1,210 @@
+<?php
+/**
+ * Adds an http based signaling server for real time collaboration.
+ *
+ * @package gutenberg
+ */
+
+/**
+ * Reads the contents of $fd and returns them unserialized.
+ *
+ * @access private
+ * @internal
+ *
+ * @param integer $fd      A file descriptor.
+ *
+ * @return array Unserialized contents of fd.
+ */
+function _gutenberg_get_contents_from_file_descriptor( $fd ) {
+	$contents_raw = stream_get_contents( $fd );
+	$result       = array();
+	if ( $contents_raw ) {
+		$result = unserialize( $contents_raw );
+	}
+	return $result;
+}
+
+
+/**
+ * Makes the file descriptor content of $fd equal to the serialization of content.
+ * Overwrites what was previously in $fd.
+ *
+ * @access private
+ * @internal
+ *
+ * @param integer $fd      A file descriptor.
+ * @param array   $content Content to be serialized and written in a file descriptor.
```

Comment:
`$fd` is not an integer.
```suggestion
 * @param resource $fd      A file descriptor.
 * @param array    $content Content to be serialized and written in a file descriptor.
```